### ISSUE TITLE
question: Span duration seems too long while in high load.

### ISSUE THREAD

**Issue by kyfk** [DataDog/dd-trace-go#958]
Our system is running on AWS ECS, and dd agents are running as a sidecar for each task.
The sqlx wrapper https://github.com/DataDog/dd-trace-go/blob/v1/contrib/jmoiron/sqlx/sql.go is used in the project as MySQL accessor.

I've executed a stress test for our system.

A part of the result is following;

<img width="636" alt="Screen Shot 0003-06-28 at 14 39 11" src="https://user-images.githubusercontent.com/12597904/123586044-2322c880-d81f-11eb-9524-b249d2b029e6.png">

Related code is below;

```
func (r *Xxxxxx) GetByID(ctx context.Context, id uint) (*xxx.Xxxxx, error) {
	span, ctx := tracer.StartSpanFromContext(ctx, "XXXXX.GetByID")
	defer span.Finish()

	var  xxx.Xxxxx
	err := r.reader.GetContext(ctx, &x, `select * from xxxxx where id=?`, id) // reader is sqlx.DB
	if err == sql.ErrNoRows {
		return nil, nil
	}
	return &x, errors.WithStack(err)
}
```

Why there's a difference in duration between sqlx wrapper's span and `tracer.StartSpanFromContext ` when a load of the task is high?
The span of `tracer.StartSpanFromContext` seems too long.

When I've seen the slow logs in AWS RDS Aurora's dashboard, the wrapper client's duration seems to collect.

`datadog.tracer.flush_duration` metrics is also long at the most same time.

### Dependencies
- gopkg.in/DataDog/dd-trace-go.v1 v1.31.1

**Comment by knusbaum** [comment]
@kyfk, Whatever starts the child span in `GetContext` may not run immediately. There may be startup operations, scheduling latency, lots of other stuff that could cause that gap, especially when there is high load.

This looks like a perfect use case for profiling. That should help you determine what is taking so long in the process.

**Comment by felixge** [comment]
I actually just ran into the same issue with an application I use for testing the profiler.

![image](https://user-images.githubusercontent.com/15000/135074941-89545d40-b00e-402a-b7d6-031159a29142.png)

My conclusion is that the current sql tracing implementation does not track time spend waiting on a new connection to be available. This can happen in two cases:

1. A new connection to the database needs to be established.
2. The internal connection pool is exhausted and the client has to wait for one of the busy connections to free up.

Further I believe this is a duplicate of #760 which has a good proposal for solving the problem, so I'm closing this.
